javascripts/controllers/FavoritesCtrl.js

```diff
@@ -1,5 +1,53 @@
 "use strict";
 
-app.controller("FavoritesCtrl", function($scope) {
-	$scope.controller = "FavoritesCtrl";
+app.controller("FavoritesCtrl", function($location, $rootScope, $scope, ContactService) {
+	
+
+	const getContacts = () => {
+		ContactService.getFavorites($rootScope.uid).then((results) => {
+			$scope.contacts = results;
+			console.log(results);
+		}).catch((err) => {
+			console.log("error in getContacts", err);
+		});
+		
+	};
+
+
+$scope.deleteContact = (userId) => {
+		ContactService.deleteContact(userId).then((results) => {
+			getContacts();
+		}).catch((err) => {
+			console.log("error in deleteContact", err);
+		});
+	};
+
+
+$scope.favoriteContacts = (contact) => {
+	let updatedContact = {};
+
+	if (!contact.favorite) {
+		updatedContact = ContactService.createContactObject(contact);
+	} else {
+		updatedContact = ContactService.createContactObject(contact);
+		updatedContact.favorite = false;
+	}
```

Comment:
So this  "update favorite" functionality isn't working for the favorites page...